includes/html/pages/alert-log.inc.php

```diff
@@ -18,5 +18,5 @@
 $vars['fromdevice'] = false;
 require_once 'includes/html/modal/alert_details.php';
 require_once 'includes/html/common/alert-log.inc.php';
-echo implode('', $common_output);
+echo htmlspecialchars(implode('', $common_output));
```

Comment:
I'm pretty sure $common_output is html.  You will need to make any fixes in `includes/html/common/alert-log.inc.php` instead.

Reply:
Thank you for your response.

I can see that $common_output has the variable $selected_min_severity in [here](https://github.com/librenms/librenms/blob/87951d28fdd3951495f59e8cf2cec4d025973209/includes/html/common/alert-log.inc.php#L122).
Which has a source [here](https://github.com/librenms/librenms/blob/87951d28fdd3951495f59e8cf2cec4d025973209/includes/html/common/alert-log.inc.php#L72)
What do you think ?

Reply:
Fix it at the source ;)

Reply:
Funny below it is cast to int (preventing xss)

Reply:
Don't forget the other $common_output ones.

Reply:
> Funny below it is cast to int (preventing xss)

This seems like the (more) correct fix

Reply:
If I understand well the cast for int is for the function array_search. but the variable $selected_min_severity will keep the string value. Because there is a concat string .= in the second assign.

```php
$selected_min_severity = '<option value="' . $_POST['min_severity'] . '" selected="selected">';
$selected_min_severity .= array_search((int) $_POST['min_severity'], $alert_severities) . '</option>';
```